src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs

```diff
@@ -170,6 +171,15 @@ public async Task<Either<ActionResult, PublicationViewModel>> UpdatePublication(
                     return Unit.Instance;
                 })
                 .OnSuccessDo(async publication =>
+                {
```

Comment:
Just a suggestion, but you could potentially also write this:

```
.OnSuccessDo(publication => ValidateThemeChange(publication, updatedPublication))
```

and:

```
        private async Task<Either<ActionResult, Unit>> ValidateThemeChange(Publication publication, PublicationSaveRequest updatedPublication)
        {
            if (publication.ThemeId != updatedPublication.ThemeId)
            {
                    var theme = await _context.Themes.FindAsync(themeId);

                    if (theme is null)
                    {
                          return ValidationActionResult(ThemeDoesNotExist);
                    }
            }

            return Unit.Instance;
        }
```

Reply:
But then I saw that you've just used the established pattern that's already there, so it's not worth doing in this one instance.

But would make a more readable chain if it was:

```
            return await _persistenceHelper
                .CheckEntityExists<Publication>(publicationId)
                .OnSuccess(_userService.CheckCanUpdatePublicationSummary)
                .OnSuccessDo(publication => ValidateTitleChange(publication, updatedPublication))
                .OnSuccessDo(publication => ValidateSupersededByChange(publication, updatedPublication))
                .OnSuccessDo(publication => ValidateThemeChange(publication, updatedPublication))
                .OnSuccessDo(blah...)
```

Anyhoo, not suggesting you make any of these changes unless you want to!

Reply:
I'll leave as-is for now.